interfaces/cython/cantera/test/utilities.py

```diff
@@ -6,47 +6,66 @@
 import tempfile
 import unittest
 import errno
+from pathlib import Path
+
+try:
+    import ruamel_yaml as yaml
+except ImportError:
+    from ruamel import yaml
+
 import cantera
 
 slow_test = unittest.skipIf(os.environ.get("CT_SKIP_SLOW", "0") == "1", "slow test")
 
+
+def load_yaml(yml_file, typ="safe"):
+    # Load YAML data from file. The YAML loader defaults to "safe".
+    yaml_ = yaml.YAML(typ=typ)
+    with open(yml_file, "rt", encoding="utf-8") as stream:
+        return yaml_.load(stream)
+
+
 class CanteraTest(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
         # Create a working directory for output files. If this is
         # an in-source test, create the directory in the root
         # test/work directory. Otherwise, create a system level
         # temporary directory
-        root_dir = os.path.abspath(os.path.join(os.path.dirname(__file__),
-                                                '..', '..', '..', '..'))
-        if os.path.exists(os.path.join(root_dir, 'SConstruct')):
-            cls.test_work_dir = os.path.join(root_dir, 'test', 'work', 'python')
+        root_dir = Path(__file__).parents[4].resolve()
+        if (root_dir / "Sconstruct").is_file():
+            cls.test_work_path = root_dir / "test" / "work" / "python"
             cls.using_tempfile = False
             try:
-                os.makedirs(cls.test_work_dir)
-            except OSError as e:
-                if e.errno == errno.EEXIST:
-                    pass
-                elif e.errno == errno.EACCES:
-                    cls.test_work_dir = tempfile.mkdtemp()
-                else:
-                    raise
+                cls.test_work_path.mkdir()
+            except FileExistsError:
+                pass
+            except FileNotFoundError:
+                cls.test_work_path = Path(tempfile.mkdtemp())
+                cls.using_tempfile = True
+            except:
+                raise
```

Comment:
This isn't needed, if the exception doesn't match it will automatically be raised.